### PR TITLE
Add --port-forward to skaffold run

### LINKED ISSUE #2949: Add port forwarding to skaffold run
We have `--tail` for logs, so we should enable `--port-forward` for `skaffold run` the same way we do it for `skaffold dev`

### PR DESCRIPTION
**Pull request by oke-py**
Fixes #2949

**Description**

Add --port-forward to skaffold run. See more details in #2949

**User facing changes**

**Before**
No --port-forward to skaffold run

**After**
Add --port-forward to skaffold run

**Next PRs.**
n/a

**Submitter Checklist**

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [unit tests](../DEVELOPMENT.md#creating-a-pr)
- [x] Mentions any output changes.
- [x] Adds documentation as needed: user docs, YAML reference, CLI reference.
- [x] Adds integration tests if needed.

**Reviewer Notes**

- [ ] The code flow looks good. 
- [ ] Unit test added.
- [ ] User facing changes look good.

**Release Notes**

```
Add --port-forward to skaffold run
```

### CODE REVIEW

**Comment by oke-py** [comment]
I'll try it.

File: integration/port_forward_test.go [hunk 1]
```diff
@@ -55,10 +55,28 @@ func TestPortForward(t *testing.T) {
 	portforward.WhiteBoxPortForwardCycle(t, kubectlCLI, ns.Name)
 }
 
-// TestPortForwardDeletePod tests that port forwarding works
+func TestRunPortForwardDeletePod(t *testing.T) {
```
Comment (balopat) [inline integration/port_forward_test.go]: nit: this is not deleting so we can remove the `Delete` from the name

File: integration/port_forward_test.go [hunk 2]
```diff
@@ -55,10 +55,28 @@ func TestPortForward(t *testing.T) {
...
+	if testing.Short() || RunOnGCP() {
+		t.Skip("skipping kind integration test")
+	}
+
+	ns, _, deleteNs := SetupNamespace(t)
+	defer deleteNs()
+
+	stop := skaffold.Run("--port-forward").InDir("examples/microservices").InNs(ns.Name).RunBackground(t)
```
Comment (balopat) [inline integration/port_forward_test.go]: you'll need to add some more flags to enable the API server: e.g. `--rpc-http-port 50051 --enable-rpc=true` - but I'd rather you use a random port like in the other test.

**Comment by oke-py** [comment]
@balopat Tests passed. Thanks for your advice!

**Comment by tvvignesh** [comment]
Hi. I noticed that this was released in v1.1.0 but tested it in Ubuntu 19.10 with latest binary but it does not seem to work. Screenshot below.

Screenshot from 2019-12-27 04-18-34

**Comment by nkubala** [comment]
@tvvignesh we decided to revert this PR for now. the behavior was non-intuitive as the process would hang after completing the deployment and forwarding the port, making it feel like it was frozen. see https://github.com/GoogleContainerTools/skaffold/issues/3415 for details

**Pull request merged**